cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -185,14 +186,26 @@ def _add_resources(self):
                 bucket=self.bucket,
                 create_lambda_roles=self._condition_create_lambda_iam_role(),
                 compute_security_groups=self.compute_security_groups,  # Empty dict if provided by the user
-                shared_storage_ids=self.shared_storage_ids,
+                shared_storage_ids=self.shared_storage_mappings,
                 shared_storage_options=self.shared_storage_options,
                 head_node_instance=self.head_node_instance,
             )
 
+        # CloudWatchDashboardSubstack
+        if self.config.is_cw_dashboard_enabled:
+            self.cloudwatch_dashboard = CWDashboardConstruct(
+                scope=self,
+                id="PclusterDashboard",
+                stack_name=self.stack_name,
+                cluster_config=self.config,
+                head_node_instance=self.head_node_instance,
+                shared_storages=self.shared_storage_mappings,
+                cw_log_group_name=self.log_group.log_group_name if self.config.is_cw_logging_enabled else None,
+            )
+
     def _add_cluster_log_group(self):
         timestamp = f"{datetime.now().strftime('%Y%m%d%H%M')}"
-        logs.CfnLogGroup(
+        self.log_group = logs.CfnLogGroup(
```

Comment:
minor: the idea is to return the resource or the resource ref and set as self attribute in the _add_resources method.
In this way is clear where/when we're adding attributes to self to be used by other methods.